File: src/main/java/io/vertigo/shiny/components/dataviz/sankey/ShinySankey.java
package io.vertigo.shiny.components.dataviz.sankey;

import java.util.List;

import io.vertigo.core.lang.Assertion;
import io.vertigo.shiny.components.ShinyComponent;

public record ShinySankey(
		String title,
		List<ShinySankeyLink> data) implements ShinyComponent {

	@Override
	public String type() {
		return "ShinySankey";
	}


	public ShinySankey {
		Assertion.check()
				.isNotBlank(title, "Title cannot be blank");
		Assertion.check()
				.isNotNull(data, "Data list cannot be null");
	}
}

File: banshee/src/models/ShinySankey.ts
import { ShinyComponent } from '../ShinyComponent';
import { ShinySankeyLink } from './ShinySankeyLink';

export class ShinySankey implements ShinyComponent {
  title: string;
  data: ShinySankeyLink[];
  type: string = 'sankey';

  constructor(title: string, data: ShinySankeyLink[]) {
    this.title = title;
    this.data = data;
  }
}
